--- src/riemann/logging.clj
+++ src/riemann/logging.clj
@@ -9,6 +9,8 @@
              ConsoleAppender
              FileAppender
              SimpleLayout)
+           (org.apache.log4j.net
+             SyslogAppender)
            (net.logstash.log4j JSONEventLayoutV0
                                JSONEventLayoutV1)
            (org.apache.log4j.spi RootLogger))
@@ -59,6 +61,7 @@
   :console         Determine if logging should happen on the console
   :console-layout  On the off-chance that someone runs riemann within runit,
                    keep the option of specifying the layout
+  :syslog          Determine whether log entries should be delivered to syslog
   :file            The file to log to. If omitted, logs to console only. If
                    provided log to that file using the default layout
   :files           A list of files to log to. If provided, a seq is expected
@@ -68,6 +71,8 @@
                    otherwise use default time based.
   Example:

+  (init {:syslog true :console true})
+    or
   (init {:console false :file \"/var/log/riemann.log\"})
     or
   (init {:console false :file \"/var/log/riemann.log\" :logsize-rotate 1000000000})"
@@ -82,11 +87,15 @@
                      (first opts)
                      (apply array-map opts))
          console (get opts :console true)
+         syslog (get opts :syslog true)
          logger (doto (Logger/getRootLogger) (.removeAllAppenders))]

     (when console
       (.addAppender logger (ConsoleAppender. (get-layout console-layout))))

+    (when syslog
+      (.addAppender logger (SyslogAppender. (get-layout :riemann) "localhost:514" (SyslogAppender/LOG_DAEMON))))
+
     (when file
       (if logsize-rotate
         (let [rolling-policy (doto (FixedWindowRollingPolicy.)
